test/integration/ads_integration_test.cc

```diff
@@ -219,6 +219,75 @@ TEST_P(AdsIntegrationTest, ClusterSharingSecretWarming) {
   test_server_->waitForGaugeEq("cluster_manager.warming_clusters", 0);
 }
 
+// Make sure two clusters with different secrets send only a single SDS request.
+// This is a regression test of #21518.
+TEST_P(AdsIntegrationTest, SecretsPausedDuringCDS) {
```

Comment:
Can you add a test for the old path with the runtime flag disabled?

Reply:
Good idea. Done.